### DISCUSSION TITLE
Adding Multiple Copies of the Same Vertex To One Mesh

### DISCUSSION THREAD

**Discussion by PixelTomato** [pokepetter/ursina#258] [Q&A] [answered]
I am attempting to create a mesh made of cubes packed together side by side but I want to hide the faces that are touching. In order to do this, I would like to iterate for every cube in the mesh, let's say 10 by 10 by 10, and check an array to see if a neighboring cube exists on each side. If a cube exists on that side, I do not want to add triangles to create a face there because the user would never be able to see it but if there is no cube, I want to add two triangles to the mesh in that location to create a square face for that side of the cube. The problem arises when I try to texture the cubes and need three different UV coordinates for each vertex meaning I need to create a copy of the vertex for each triangle I add. I cannot seem to figure out how to do this because every time I attempt to add a vertex that already exists to the mesh, any triangles that attempt to use it do not render.

I would be very thankful for any suggestions anyone may have. Thanks in advance!

**Comment by pokepetter** [comment, marked as answer]
I think you must have messed up the triangle indices. Duplicate vertices shouldn't matter. However, if you need every vertex to be unique anyway, you can just add the vertices in order and you won't need the triangles. That's what I did when making the same thing.

**Reply by PixelTomato** [reply]
Thank you for your help! I discovered that I was creating hundreds of triangles attached to the same vertices because I was not adding an offset to the tuple containing the triangle vertex indexes. I now apply an offset to each element of the tuple through a custom function because I could not figure out a way to make `numpy.add()` return a format that Ursina would accept. The mesh gets constructed successfully as well as in proper order so that UV mappings can be applied easily.
